ethexe/network/src/lib.rs

```diff
@@ -447,51 +356,94 @@ impl NetworkEventLoop {
             BehaviourEvent::Gossipsub(_) => {}
             //
             BehaviourEvent::DbSync(db_sync::Event::ExternalValidation(validating_response)) => {
-                let _res = self
-                    .external_tx
-                    .send(NetworkReceiverEvent::ExternalValidation(
-                        validating_response,
-                    ));
+                return Some(NetworkEvent::ExternalValidation(validating_response));
             }
             BehaviourEvent::DbSync(db_sync::Event::RequestSucceed {
                 request_id: _,
                 response,
             }) => {
-                let _res = self
-                    .external_tx
-                    .send(NetworkReceiverEvent::DbResponse(Ok(response)));
+                return Some(NetworkEvent::DbResponse(Ok(response)));
             }
             BehaviourEvent::DbSync(db_sync::Event::RequestFailed {
                 request_id: _,
                 error,
             }) => {
-                let _res = self
-                    .external_tx
-                    .send(NetworkReceiverEvent::DbResponse(Err(error)));
+                return Some(NetworkEvent::DbResponse(Err(error)));
             }
             BehaviourEvent::DbSync(_) => {}
         }
+
+        None
     }
 
-    fn handle_network_rx_event(&mut self, event: NetworkSenderEvent) {
-        match event {
-            NetworkSenderEvent::PublishMessage { data } => {
-                if let Err(e) = self
-                    .swarm
-                    .behaviour_mut()
-                    .gossipsub
-                    .publish(gpu_commitments_topic(), data)
-                {
-                    log::debug!("gossipsub publishing failed: {e}")
-                }
-            }
-            NetworkSenderEvent::RequestDbData(request) => {
-                self.swarm.behaviour_mut().db_sync.request(request);
-            }
-            NetworkSenderEvent::RequestValidated(res) => {
-                self.swarm.behaviour_mut().db_sync.request_validated(res);
+    pub fn local_peer_id(&self) -> PeerId {
+        *self.swarm.local_peer_id()
+    }
+
+    pub fn score_handle(&self) -> peer_score::Handle {
+        self.swarm.behaviour().peer_score.handle()
+    }
+
+    pub fn publish_message(&mut self, data: Vec<u8>) {
```

Comment:
mb intoduce methods with `impl Encode` and `Vec<u8>`?

```suggestion
    pub fn publish_message(&mut self, data: impl Encode) {
```

```rust
pub fn publish_message(&mut self, data: Vec<u8>)  {}
pub fn publish_network_message(&mut self, message: impl Encode)  {}
```

Reply:
to be done in separate pr. I do want to make all of networking methods typed